{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Задача\n",
    "\n",
    "- Прочитать файл\n",
    "- Посчитать количество последовательностей в fasta файле;\n",
    "- Добавить рассчет статистики, которую писали для строк для каждой последовательности, присутсвующей в fasta файле;\n",
    "- Найти в последовательности стар и стоп кодоны и записать их координанты в последовательности;\n",
    "- если это ДНК: сделать транскрипцию, а затем трансляцию, причем для случая 5'-3' и 3'-5'. В случае если это РНК, сделать только трансляцию;\n",
    "- Основываясь на таблице аминокислот посчитать заряд полученной амминокислотной последовательности;\n",
    "- Добавить возможность записи всей статистики в файл для пользователя, а также записывать в файл результат транскрипиции, трансляции;\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">NC_003322.1:14176-15321 Vombatus ursinus mitochondrion, complete genome\n",
      "ATGATTAACCTACGCAAAACACATCCCCTTATAAAAATCGTCAATGAAGCATTCATCGACCTACCCACACCCTCCAATATCTCCGCCTGATGAAATTTTGGATCACTATTAGGAATCTGCCTTATCATACAAATCCTAACAGGCCTATTCCTAGCCATACATTACACCTCCGATACCCTAACAGCCTTCTCTTCAGTAGCCCATATCTGCCGAGATGTGAATCACGGCTGACTCATCCGCAACCTCCACGCCAACGGAGCGTCTATATTCTTCATATGCCTATACCTCCACATTGGCCGAGGAATCTATTATGGCTCCTACCTCTACAAAGAAACATGAAACATCGGAGTATTTCTTCTACTCACAGTTATAGCAACTGCTTTCGTTGGCTATGTACTCCCATGAGGACAAATATCCTTCTGAGGTGCAACCGTAATTACCAACTTATTATCAGCCATCCCTTACGTAGGCACCACCCTAGTAGAATGAATTTGAGGTGGATTCTCCGTAGACAAAGCCACACTGACCCGATTCTTCGCCTTCCATTTTATCCTACCCTTCATTGTCACAGCACTAGCTATCGTTCACCTACTATTCCTTCATGAAACAGGCTCAAACAACCCCTCAGGAATTAACCCCGACGCAGACAAAATTCCCTTCCACCCCTACTACACTACCAAAGACATTATAGGTGCAATCCTAATAATCCTTGCCCTCCTACTACTTACCCTATTCTCACCCGACATGTTAGGAGACCCAGACAACTTCTCCCCTGCCAACCCCCTCAGCACACCACCCCACATCAAACCAGAATGATATTTCCTATTCGCCTACGCTATTCTCCGATCAATCCCAAATAAACTGGGAGGAGTACTAGCCCTACTAGCATCCATCCTAATCCTCCTAGTCATCCCATTCCTGCATACATCCAAACAGCGCAGTATAATATTTCGACCCATCTCCCAAACCCTCTTCTGAATTTTAATCGCAAACCTCCTCATCCTAACCTGAATCGGAGGACAACCTGTAGAACAACCTTACATTATCATCGGACAAACTGCATCCATCACATACTTCCTCCTAATCCTCATTCTAATACCCGCAGCAGGACTATTTGAAAACTACATACTAAAACCTAAATGAAGA\n",
      "\n",
      "----------------------------------------------------------------------------------------\n",
      "STATS:\n",
      "Initial sequence: ATGATTAACCTACGCAAAACACATCCCCTTATAAAAATCGTCAATGAAGCATTCATCGACCTACCCACACCCTCCAATATCTCCGCCTGATGAAATTTTGGATCACTATTAGGAATCTGCCTTATCATACAAATCCTAACAGGCCTATTCCTAGCCATACATTACACCTCCGATACCCTAACAGCCTTCTCTTCAGTAGCCCATATCTGCCGAGATGTGAATCACGGCTGACTCATCCGCAACCTCCACGCCAACGGAGCGTCTATATTCTTCATATGCCTATACCTCCACATTGGCCGAGGAATCTATTATGGCTCCTACCTCTACAAAGAAACATGAAACATCGGAGTATTTCTTCTACTCACAGTTATAGCAACTGCTTTCGTTGGCTATGTACTCCCATGAGGACAAATATCCTTCTGAGGTGCAACCGTAATTACCAACTTATTATCAGCCATCCCTTACGTAGGCACCACCCTAGTAGAATGAATTTGAGGTGGATTCTCCGTAGACAAAGCCACACTGACCCGATTCTTCGCCTTCCATTTTATCCTACCCTTCATTGTCACAGCACTAGCTATCGTTCACCTACTATTCCTTCATGAAACAGGCTCAAACAACCCCTCAGGAATTAACCCCGACGCAGACAAAATTCCCTTCCACCCCTACTACACTACCAAAGACATTATAGGTGCAATCCTAATAATCCTTGCCCTCCTACTACTTACCCTATTCTCACCCGACATGTTAGGAGACCCAGACAACTTCTCCCCTGCCAACCCCCTCAGCACACCACCCCACATCAAACCAGAATGATATTTCCTATTCGCCTACGCTATTCTCCGATCAATCCCAAATAAACTGGGAGGAGTACTAGCCCTACTAGCATCCATCCTAATCCTCCTAGTCATCCCATTCCTGCATACATCCAAACAGCGCAGTATAATATTTCGACCCATCTCCCAAACCCTCTTCTGAATTTTAATCGCAAACCTCCTCATCCTAACCTGAATCGGAGGACAACCTGTAGAACAACCTTACATTATCATCGGACAAACTGCATCCATCACATACTTCCTCCTAATCCTCATTCTAATACCCGCAGCAGGACTATTTGAAAACTACATACTAAAACCTAAATGAAGA\n",
      "This is a DNA sequence\n",
      "Mean value = 286.5\n",
      "Length = 1146\n",
      "{'A': 340, 'T': 296, 'G': 144, 'C': 366}\n",
      "GC - composition = 44.50261780104712%\n",
      "Second DNA chain: TCTTCATTTAGGTTTTAGTATGTAGTTTTCAAATAGTCCTGCTGCGGGTATTAGAATGAGGATTAGGAGGAAGTATGTGATGGATGCAGTTTGTCCGATGATAATGTAAGGTTGTTCTACAGGTTGTCCTCCGATTCAGGTTAGGATGAGGAGGTTTGCGATTAAAATTCAGAAGAGGGTTTGGGAGATGGGTCGAAATATTATACTGCGCTGTTTGGATGTATGCAGGAATGGGATGACTAGGAGGATTAGGATGGATGCTAGTAGGGCTAGTACTCCTCCCAGTTTATTTGGGATTGATCGGAGAATAGCGTAGGCGAATAGGAAATATCATTCTGGTTTGATGTGGGGTGGTGTGCTGAGGGGGTTGGCAGGGGAGAAGTTGTCTGGGTCTCCTAACATGTCGGGTGAGAATAGGGTAAGTAGTAGGAGGGCAAGGATTATTAGGATTGCACCTATAATGTCTTTGGTAGTGTAGTAGGGGTGGAAGGGAATTTTGTCTGCGTCGGGGTTAATTCCTGAGGGGTTGTTTGAGCCTGTTTCATGAAGGAATAGTAGGTGAACGATAGCTAGTGCTGTGACAATGAAGGGTAGGATAAAATGGAAGGCGAAGAATCGGGTCAGTGTGGCTTTGTCTACGGAGAATCCACCTCAAATTCATTCTACTAGGGTGGTGCCTACGTAAGGGATGGCTGATAATAAGTTGGTAATTACGGTTGCACCTCAGAAGGATATTTGTCCTCATGGGAGTACATAGCCAACGAAAGCAGTTGCTATAACTGTGAGTAGAAGAAATACTCCGATGTTTCATGTTTCTTTGTAGAGGTAGGAGCCATAATAGATTCCTCGGCCAATGTGGAGGTATAGGCATATGAAGAATATAGACGCTCCGTTGGCGTGGAGGTTGCGGATGAGTCAGCCGTGATTCACATCTCGGCAGATATGGGCTACTGAAGAGAAGGCTGTTAGGGTATCGGAGGTGTAATGTATGGCTAGGAATAGGCCTGTTAGGATTTGTATGATAAGGCAGATTCCTAATAGTGATCCAAAATTTCATCAGGCGGAGATATTGGAGGGTGTGGGTAGGTCGATGAATGCTTCATTGACGATTTTTATAAGGGGATGTGTTTTGCGTAGGTTAATCAT\n",
      "----------------------------------------------------------------------------------------\n",
      "5'-3' after transcription: UACUAAUUGGAUGCGUUUUGUGUAGGGGAAUAUUUUUAGCAGUUACUUCGUAAGUAGCUGGAUGGGUGUGGGAGGUUAUAGAGGCGGACUACUUUAAAACCUAGUGAUAAUCCUUAGACGGAAUAGUAUGUUUAGGAUUGUCCGGAUAAGGAUCGGUAUGUAAUGUGGAGGCUAUGGGAUUGUCGGAAGAGAAGUCAUCGGGUAUAGACGGCUCUACACUUAGUGCCGACUGAGUAGGCGUUGGAGGUGCGGUUGCCUCGCAGAUAUAAGAAGUAUACGGAUAUGGAGGUGUAACCGGCUCCUUAGAUAAUACCGAGGAUGGAGAUGUUUCUUUGUACUUUGUAGCCUCAUAAAGAAGAUGAGUGUCAAUAUCGUUGACGAAAGCAACCGAUACAUGAGGGUACUCCUGUUUAUAGGAAGACUCCACGUUGGCAUUAAUGGUUGAAUAAUAGUCGGUAGGGAAUGCAUCCGUGGUGGGAUCAUCUUACUUAAACUCCACCUAAGAGGCAUCUGUUUCGGUGUGACUGGGCUAAGAAGCGGAAGGUAAAAUAGGAUGGGAAGUAACAGUGUCGUGAUCGAUAGCAAGUGGAUGAUAAGGAAGUACUUUGUCCGAGUUUGUUGGGGAGUCCUUAAUUGGGGCUGCGUCUGUUUUAAGGGAAGGUGGGGAUGAUGUGAUGGUUUCUGUAAUAUCCACGUUAGGAUUAUUAGGAACGGGAGGAUGAUGAAUGGGAUAAGAGUGGGCUGUACAAUCCUCUGGGUCUGUUGAAGAGGGGACGGUUGGGGGAGUCGUGUGGUGGGGUGUAGUUUGGUCUUACUAUAAAGGAUAAGCGGAUGCGAUAAGAGGCUAGUUAGGGUUUAUUUGACCCUCCUCAUGAUCGGGAUGAUCGUAGGUAGGAUUAGGAGGAUCAGUAGGGUAAGGACGUAUGUAGGUUUGUCGCGUCAUAUUAUAAAGCUGGGUAGAGGGUUUGGGAGAAGACUUAAAAUUAGCGUUUGGAGGAGUAGGAUUGGACUUAGCCUCCUGUUGGACAUCUUGUUGGAAUGUAAUAGUAGCCUGUUUGACGUAGGUAGUGUAUGAAGGAGGAUUAGGAGUAAGAUUAUGGGCGUCGUCCUGAUAAACUUUUGAUGUAUGAUUUUGGAUUUACUUCU\n",
      "Indexes of start codons: [162, 282, 318, 324, 462, 666, 669, 831, 1038, 1071, 1122]\n",
      "Indexes of stop codons: [3, 36, 54, 78, 114, 123, 132, 204, 234, 291, 303, 342, 375, 435, 456, 489, 549, 561, 579, 630, 651, 672, 684, 696, 705, 801, 837, 849, 891, 897, 909, 957, 978, 984, 999, 1011, 1041, 1044, 1047, 1056, 1065, 1083, 1089]\n",
      "\n",
      "3'-5' after transcription: AGAAGUAAAUCCAAAAUCAUACAUCAAAAGUUUAUCAGGACGACGCCCAUAAUCUUACUCCUAAUCCUCCUUCAUACACUACCUACGUCAAACAGGCUACUAUUACAUUCCAACAAGAUGUCCAACAGGAGGCUAAGUCCAAUCCUACUCCUCCAAACGCUAAUUUUAAGUCUUCUCCCAAACCCUCUACCCAGCUUUAUAAUAUGACGCGACAAACCUACAUACGUCCUUACCCUACUGAUCCUCCUAAUCCUACCUACGAUCAUCCCGAUCAUGAGGAGGGUCAAAUAAACCCUAACUAGCCUCUUAUCGCAUCCGCUUAUCCUUUAUAGUAAGACCAAACUACACCCCACCACACGACUCCCCCAACCGUCCCCUCUUCAACAGACCCAGAGGAUUGUACAGCCCACUCUUAUCCCAUUCAUCAUCCUCCCGUUCCUAAUAAUCCUAACGUGGAUAUUACAGAAACCAUCACAUCAUCCCCACCUUCCCUUAAAACAGACGCAGCCCCAAUUAAGGACUCCCCAACAAACUCGGACAAAGUACUUCCUUAUCAUCCACUUGCUAUCGAUCACGACACUGUUACUUCCCAUCCUAUUUUACCUUCCGCUUCUUAGCCCAGUCACACCGAAACAGAUGCCUCUUAGGUGGAGUUUAAGUAAGAUGAUCCCACCACGGAUGCAUUCCCUACCGACUAUUAUUCAACCAUUAAUGCCAACGUGGAGUCUUCCUAUAAACAGGAGUACCCUCAUGUAUCGGUUGCUUUCGUCAACGAUAUUGACACUCAUCUUCUUUAUGAGGCUACAAAGUACAAAGAAACAUCUCCAUCCUCGGUAUUAUCUAAGGAGCCGGUUACACCUCCAUAUCCGUAUACUUCUUAUAUCUGCGAGGCAACCGCACCUCCAACGCCUACUCAGUCGGCACUAAGUGUAGAGCCGUCUAUACCCGAUGACUUCUCUUCCGACAAUCCCAUAGCCUCCACAUUACAUACCGAUCCUUAUCCGGACAAUCCUAAACAUACUAUUCCGUCUAAGGAUUAUCACUAGGUUUUAAAGUAGUCCGCCUCUAUAACCUCCCACACCCAUCCAGCUACUUACGAAGUAACUGCUAAAAAUAUUCCCCUACACAAAACGCAUCCAAUUAGUA\n",
      "Indexes of start codons: [117, 273, 636, 663, 678, 711, 750, 795, 948]\n",
      "Indexes of stop codons: [204, 288, 924, 930, 972, 1050, 1068, 1101]\n",
      "\n",
      "----------------------------------------------------------------------------------------\n",
      "5'-3' after transcription and translation: YendLDAFCVGEYFendQLLRKendLDGCGRLendRRTTLKPSDNPendTEendYVendDCPDKDRYVMWRLWDCRKRSHRVendTALHLVPTEendALEVRLPRRYKKYTDMEVendPAPendIIPRMEMFLCTLendPHKEDECQYRendRKQPIHEGTPVYRKTPRWHendWLNNSRendGMHPWWDHLTendTPPKRHLFRCDWAKKRKVKendDGKendQCRDRendQVDDKEVLCPSLLGSPendLGLRLFendGKVGMMendWFLendYPRendDYendEREDDEWDKSGLYNPLGLLKRGRLGESCGGVendFGLTIKDKRMRendEASendGLFDPPHDRDDRRendDendEDQendGKDVCRFVASYYKAGendRVWEKTendNendRLEEendDWTendPPVGHLVGMendendendPVendRRendCMKEDendEendDYGRRPDKLLMYDFGFT\n",
      "Charge of amino sequence: 23\n",
      "\n",
      "5'-3' after transcription and translation: RSKSKIIHQKFIRTTPIILLLILLHTLPTSNRLLLHSNKMSNRRLSPILLLQTLILSLLPNPLPSFIIendRDKPTYVLTLLILLILPTIIPIMRRVKendTLTSLLSHPLILYSKTKLHPTTRLPQPSPLQQTQRIVQPTLIPFIILPFLIILTWILQKPSHHPHLPLKQTQPQLRTPQQTRTKYFLIIHLLSITTLLLPILFYLPLLSPVTPKQMPLRWSLSKMIPPRMHSLPTIIQPLMPTWSLPINRSTLMYRLLSSTILTLIFFMRLQSTKKHLHPRYYLRSRLHLHIRILLISARQPHLQRLLSRHendVendSRLYPMTSLPTIPendPPHYIPILIRTILNILFRLRIITRFendSSPPLendPPTPIQLLTKendLLKIFPYTKRIQL\n",
      "Charge of amino sequence: 64\n",
      "\n",
      "----------------------------------------------------------------------------------------\n",
      "\n",
      "Number of sequences in fasta file: 1\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "import os\n",
    "path_to_code = os.getcwd()\n",
    "path_to_file = os.path.join(path_to_code, os.pardir, 'lections', 'files_and_images', 'wombat_mit.fasta')\n",
    "   \n",
    "#---------------------------------------------------------------    \n",
    "    \n",
    "def read_fasta_file(inp_fasta_file):\n",
    "    name, seq = '', []\n",
    "    for line in inp_fasta_file:\n",
    "        line = line.rstrip()\n",
    "        if line.startswith(\">\"):\n",
    "            if name:\n",
    "                yield(name, ''.join(seq))\n",
    "            name, seq = line, []\n",
    "        else: \n",
    "            seq.append(line)\n",
    "    if name: \n",
    "        yield(name, ''.join(seq))\n",
    "        \n",
    "#---------------------------------------------------------------\n",
    "        \n",
    "def nucl_seq_type(nucl_seq):\n",
    "    par = False\n",
    "    if 'T' in list(nucl_seq): par = True\n",
    "    return par\n",
    "\n",
    "def nucl_seq_mean_value(nucl_seq):\n",
    "    mean_value = (list(nucl_seq).count('A') + list(nucl_seq).count('G') + list(nucl_seq).count('C') + list(nucl_seq).count('T') + list(nucl_seq).count('U')) / 4\n",
    "    return mean_value\n",
    "\n",
    "def nucl_seq_count(nucl_seq):\n",
    "    dict_of_nucl = {}\n",
    "    if nucl_seq_type(nucl_seq):\n",
    "        for elem in list('ATGC'):\n",
    "            dict_of_nucl.update([(elem, list(nucl_seq).count(elem))])\n",
    "    else:\n",
    "        for elem in list('AUGC'):\n",
    "            dict_of_nucl.update([(elem, list(nucl_seq).count(elem))])\n",
    "    return dict_of_nucl\n",
    "\n",
    "def nucl_seq_gc(nucl_seq):\n",
    "    dict_of_nucl = nucl_seq_count(nucl_seq)\n",
    "    gc_composition = ((dict_of_nucl['G'] + dict_of_nucl['C']) / len(nucl_seq)) * 100\n",
    "    return gc_composition\n",
    "\n",
    "def nucl_seq_complement(nucl_seq):\n",
    "    nucl_seq_comp = []\n",
    "    for elem in nucl_seq:\n",
    "        if elem == 'A': nucl_seq_comp.append('T')\n",
    "        if elem == 'T': nucl_seq_comp.append('A')\n",
    "        if elem == 'G': nucl_seq_comp.append('C')\n",
    "        if elem == 'C': nucl_seq_comp.append('G')\n",
    "    return ''.join(nucl_seq_comp)\n",
    "\n",
    "def nucl_seq_second_chain(nucl_seq):\n",
    "    if nucl_seq_type(nucl_seq): nucl_seq_sec = nucl_seq_complement(nucl_seq)[ : :-1]\n",
    "    else: nucl_seq_sec = 'Not a DNA'\n",
    "    return nucl_seq_sec\n",
    "\n",
    "def nucl_seq_stat(nucl_seq):\n",
    "    print('STATS:')\n",
    "    print(f'Initial sequence: {nucl_seq}')\n",
    "    if nucl_seq_type(nucl_seq): print('This is a DNA sequence')\n",
    "    else: print('This is a RNA sequence')\n",
    "    print(f'Mean value = {nucl_seq_mean_value(nucl_seq)}')\n",
    "    print(f'Length = {len(nucl_seq)}')\n",
    "    print(nucl_seq_count(nucl_seq))\n",
    "    print(f'GC - composition = {nucl_seq_gc(nucl_seq)}%')\n",
    "    print(f'Second DNA chain: {nucl_seq_second_chain(nucl_seq)}')\n",
    "    \n",
    "def nucl_seq_stat_to_dic(nucl_seq):\n",
    "    seq_stat_dic = {}\n",
    "    seq_stat_dic['seq'] = nucl_seq\n",
    "    if nucl_seq_type(nucl_seq): seq_stat_dic['type'] = 'DNA'\n",
    "    else: seq_stat_dic['type'] = 'RNA'\n",
    "    seq_stat_dic['mean'] = nucl_seq_mean_value(nucl_seq)\n",
    "    seq_stat_dic['length'] = len(nucl_seq)\n",
    "    seq_stat_dic['count'] = nucl_seq_count(nucl_seq)\n",
    "    seq_stat_dic['gs'] = nucl_seq_gc(nucl_seq)\n",
    "    seq_stat_dic['sec_chain'] = nucl_seq_second_chain(nucl_seq)    \n",
    "    return seq_stat_dic\n",
    "\n",
    "def double_nucl_seq_stat_to_dic(first_nucl_seq):\n",
    "    second_nucl_seq = nucl_seq_second_chain(first_nucl_seq)\n",
    "    double_seq_stat_dic = {}\n",
    "    double_seq_stat_dic['5\\'-3\\''] = nucl_seq_stat_to_dic(first_nucl_seq)\n",
    "    double_seq_stat_dic['3\\'-5\\''] = nucl_seq_stat_to_dic(second_nucl_seq)  \n",
    "    return double_seq_stat_dic        \n",
    "    \n",
    "#---------------------------------------------------------------   \n",
    "\n",
    "def nucl_seq_transcription(nucl_seq):\n",
    "    if nucl_seq_type(nucl_seq): \n",
    "        nucl_seq_transcr = nucl_seq_complement(nucl_seq).replace('T','U')\n",
    "    else: nucl_seq_transcr = 'Need DNA for transcription'\n",
    "    return nucl_seq_transcr\n",
    "\n",
    "def nucl_seq_translation(nucl_seq):\n",
    "    amino_seq = []\n",
    "    if not nucl_seq_type(nucl_seq):\n",
    "        for i in range(0, len(nucl_seq) - 3, 3):\n",
    "            amino_seq.append(gene_code_dic[nucl_seq[i:i+3]])\n",
    "    else: amino_seq.append('Need RNA for translation')\n",
    "    return ''.join(amino_seq)    \n",
    "\n",
    "def nucl_seq_start_codons_indexes(nucl_seq):\n",
    "    regexp_start_codons = re.compile('AUG')\n",
    "    start_codons_indexes = []\n",
    "    index = 0\n",
    "    if regexp_start_codons.search(nucl_seq):\n",
    "        while regexp_start_codons.search(nucl_seq[index:]):\n",
    "            if regexp_start_codons.search(nucl_seq[index:]).start() % 3 == 0: \n",
    "                start_codons_indexes.append(index + regexp_start_codons.search(nucl_seq[index:]).start())\n",
    "                index += regexp_start_codons.search(nucl_seq[index:]).end()\n",
    "            elif (regexp_start_codons.search(nucl_seq[index:]).start() - 1) % 3 == 0: index += regexp_start_codons.search(nucl_seq[index:]).start() + 2\n",
    "            else: index += regexp_start_codons.search(nucl_seq[index:]).start() + 1\n",
    "    return start_codons_indexes\n",
    "\n",
    "def nucl_seq_stop_codons_indexes(nucl_seq):\n",
    "    regexp_stop_codons = re.compile('UA[AG]|UGA')\n",
    "    stop_codons_indexes = []\n",
    "    index = 0\n",
    "    if regexp_stop_codons.search(nucl_seq):\n",
    "        while regexp_stop_codons.search(nucl_seq[index:]):\n",
    "            if regexp_stop_codons.search(nucl_seq[index:]).start() % 3 == 0: \n",
    "                stop_codons_indexes.append(index + regexp_stop_codons.search(nucl_seq[index:]).start())\n",
    "                index += regexp_stop_codons.search(nucl_seq[index:]).end()\n",
    "            elif (regexp_stop_codons.search(nucl_seq[index:]).start() - 1) % 3 == 0: index += regexp_stop_codons.search(nucl_seq[index:]).start() + 2\n",
    "            else: index += regexp_stop_codons.search(nucl_seq[index:]).start() + 1\n",
    "    return stop_codons_indexes\n",
    "\n",
    "def charge_of_amino_seq(amino_seq):\n",
    "    charge = 0\n",
    "    for char in amino_seq:\n",
    "        if char in amino_acids_charge_dic.keys():\n",
    "            charge += amino_acids_charge_dic[char]\n",
    "    return charge\n",
    "\n",
    "#---------------------------------------------------------------  \n",
    "\n",
    "gene_code_dic = {'UUU': 'F', 'UUC': 'F', 'UUA': 'L', 'UUG': 'L', 'UCU': 'S', 'UCC': 'S', 'UCA': 'S', 'UCG': 'S', 'UAU': 'Y', 'UAC': 'Y', 'UAA': 'end', 'UAG': 'end', \\\n",
    "                'UGU' : 'C', 'UGC': 'C', 'UGA': 'end', 'UGG': 'W', 'CUU': 'L', 'CUC': 'L', 'CUA': 'L', 'CUG': 'L', 'CCU': 'P', 'CCC': 'P', 'CCA': 'P', 'CCG': 'P', \\\n",
    "                'CAU': 'H', 'CAC': 'H', 'CAA': 'Q', 'CAG': 'Q', 'CGU': 'R', 'CGC': 'R', 'CGA': 'R', 'CGG': 'R', 'AUU': 'I', 'AUC': 'I', 'AUA': 'I', 'AUG': 'M', \\\n",
    "                'ACU': 'T', 'ACC': 'T', 'ACA': 'T', 'ACG': 'T', 'AAU': 'N', 'AAC': 'N', 'AAA': 'K', 'AAG': 'K', 'AGU': 'S', 'AGC': 'S', 'AGA': 'R', 'AGG': 'R', \\\n",
    "                'GUU': 'V', 'GUC': 'V', 'GUA': 'V', 'GUG': 'V', 'GCU': 'A', 'GCC': 'A', 'GCA': 'A', 'GCG': 'A', 'GAU': 'D', 'GAC': 'D', 'GAA': 'E', 'GAG': 'E', \\\n",
    "                'GGU': 'G', 'GGC': 'G', 'GGA': 'G', 'GGG': 'G'}\n",
    "\n",
    "amino_acids_charge_dic = {'D': -1, 'E': -1, 'K': 1, 'R': 1, 'H': 1}\n",
    "\n",
    "#---------------------------------------------------------------  \n",
    "\n",
    "with open(path_to_file, 'r') as inp_fasta_file:\n",
    "    par = 0\n",
    "    for name, seq in read_fasta_file(inp_fasta_file):\n",
    "        par += 1\n",
    "        print(name, seq, sep = '\\n', end = '\\n\\n')\n",
    "        print('----------------------------------------------------------------------------------------')\n",
    "        nucl_seq_stat(seq)\n",
    "        print('----------------------------------------------------------------------------------------')\n",
    "        \n",
    "        if nucl_seq_type(seq):\n",
    "            sec_seq = nucl_seq_second_chain(seq)\n",
    "            first_seq_transcripted = nucl_seq_transcription(seq)\n",
    "            print(f'5\\'-3\\' after transcription: {first_seq_transcripted}')\n",
    "            start_codons_indexes = nucl_seq_start_codons_indexes(first_seq_transcripted)\n",
    "            stop_codons_indexes = nucl_seq_stop_codons_indexes(first_seq_transcripted)\n",
    "            print(f'Indexes of start codons: {start_codons_indexes}')\n",
    "            print(f'Indexes of stop codons: {stop_codons_indexes}', end = '\\n\\n')    \n",
    "            \n",
    "            second_seq_transcripted = nucl_seq_transcription(sec_seq)\n",
    "            print(f'3\\'-5\\' after transcription: {second_seq_transcripted}')\n",
    "            start_codons_indexes = nucl_seq_start_codons_indexes(second_seq_transcripted)\n",
    "            stop_codons_indexes = nucl_seq_stop_codons_indexes(second_seq_transcripted)\n",
    "            print(f'Indexes of start codons: {start_codons_indexes}')\n",
    "            print(f'Indexes of stop codons: {stop_codons_indexes}', end = '\\n\\n')    \n",
    "            \n",
    "            print('----------------------------------------------------------------------------------------')\n",
    "            first_seq_transcripted_translated = nucl_seq_translation(first_seq_transcripted)\n",
    "            print(f'5\\'-3\\' after transcription and translation: {first_seq_transcripted_translated}')\n",
    "            print(f'Charge of amino sequence: {charge_of_amino_seq(first_seq_transcripted_translated)}', end = '\\n\\n')\n",
    "            second_seq_transcripted_translated = nucl_seq_translation(second_seq_transcripted)\n",
    "            print(f'5\\'-3\\' after transcription and translation: {second_seq_transcripted_translated}')\n",
    "            print(f'Charge of amino sequence: {charge_of_amino_seq(second_seq_transcripted_translated)}', end = '\\n\\n')\n",
    "            \n",
    "    print('----------------------------------------------------------------------------------------')  \n",
    "    print(f'\\nNumber of sequences in fasta file: {par}')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
