{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Задача 1\n",
    "В файле gsm_list.txt содержатся идентификаторы GEO NCBI (идентификаторы вида GSM#####). Вам необходимо подгрузить html-страницу с информацией по каждому эксперименту. Например, https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM5176077. Обратите внимание на то, как устроена URL-ссылка. Ознакомтесь со страницей и имеющейся на ней информацией. Напишите программу, которая ищет соответствие серии экспериментов и эксперимента. Результат запищите ввиде таблицы, состоящей из 2 колонок.\n",
    "\n",
    "\\* В базе данных GEO NCBI некоторые из серий экспермиентов являются суперсериями, т. е. включают в себя другие серии экспериментов. Информацию об этом можно найти на соответствующих им страницам. Модифицируйте вашу программу так, чтобы идентификаторы суперсерий игнорировались"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.01 Transitional//EN\">\n",
      "\n",
      "<html>\n",
      "<head>\n",
      "<style type=\"text/css\">\n",
      "      a { text-decoration: none; }\n",
      "\t.bordergc {background-color: #6699CC;}\n",
      "\t.bordergd {background-color: #B6C7E5;}\n",
      "\t.borderge {background-color: #EEF3FB;}\n",
      "\t.bordergf {background-color: #FFFFFF;}\n",
      "\t.bordergg {background-color: #CCCCCC;}\n",
      "      .small8b { font-size:8pt;\n",
      "                font-family: ariel,helvetica,sans-serif;\n",
      "                color:#6633cc;\n",
      "              }\n",
      "      .small8db { font-size:8pt;\n",
      "                font-family: ariel,helvetica,sans-serif;\n",
      "                color:#4411aa;\n",
      "              }\n",
      "\n",
      "    </style>\n",
      "<meta content=\"text/html; charset=utf-8\" http-equiv=\"Content-Type\"/>\n",
      "<meta content=\"NCBI GEO Gene Expression Omnibus microarray oligonucleotide array SAGE\" name=\"keywords\"/>\n",
      "<meta content=\"NCBI's Gene Expression Omnibus (GEO) is a public archive and resource for gene expression data.\" name=\"description\"/>\n",
      "<meta content=\"geo\" name=\"ncbi_app\"/>\n",
      "<meta content=\"full\" name=\"ncbi_pdid\"/>\n",
      "<meta content=\"0751F9451405CB110000000000000001\" name=\"ncbi_phid\"/>\n",
      "<meta content=\"0751F9451405CB11_0000SID\" name=\"ncbi_sessionid\"/>\n",
      "<title>\n",
      "    GEO Accession viewer\n",
      "    </title>\n",
      "<link href=\"/corehtml/ncbi.css\" rel=\"stylesheet\"/>\n",
      "<!-- GEO_SCRIPT -->\n",
      "<script language=\"JavaScript1.2\" src=\"/coreweb/javascript/imagemouseover.js\"></script>\n",
      "<script language=\"JavaScript1.2\" src=\"/coreweb/javascript/show_message.js\"></script>\n",
      "<script src=\"/corehtml/jsutils/utils.1.js\" type=\"text/javascript\"></script>\n",
      "<script src=\"/corehtml/jsutils/remote_data_provider.1.js\" type=\"text/javascript\"></script>\n",
      "<script language=\"JavaScript1.2\" src=\"/geo/js/help_def_messages.js\"></script>\n",
      "<script type=\"text/javascript\">\n",
      "    window.onload = function () {\n",
      "        //old smaller style banner\n",
      "        display_global_alert = false;\n",
      "        \n",
      "        jQuery.getScript(\"/core/alerts/alerts.js\", function () {\n",
      "            galert(['#galerts_table','body > *:nth-child(1)'])\n",
      "        });\n",
      "    }\n",
      "</script>\n",
      "<link href=\"../info/geo_style.css\" rel=\"STYLESHEET\" type=\"text/css\"/>\n",
      "<link href=\"acc.css\" rel=\"stylesheet\" type=\"text/css\">\n",
      "<script language=\"Javascript\">\n",
      "\n",
      "  function OnFormFieldChange()\n",
      "  {\n",
      "    var view = document.getElementById(\"view\");\n",
      "\n",
      "    if(document.getElementById(\"ViewOptions\").form.value == 'html')\n",
      "    {\n",
      "        view.remove(3);\n",
      "        view.remove(2);\n",
      "    }\n",
      "    else\n",
      "    {\n",
      "        var NewOption = document.createElement(\"OPTION\");\n",
      "\n",
      "        NewOption.text = \"Full\";\n",
      "        NewOption.value = \"full\";\n",
      "\n",
      "        try\n",
      "        {\n",
      "            view.add(NewOption, null);\n",
      "        }\n",
      "        catch(ex)\n",
      "        {\n",
      "            view.add(NewOption);\n",
      "        }\n",
      "\n",
      "        NewOption = document.createElement(\"OPTION\");\n",
      "\n",
      "        NewOption.text = \"Data\";\n",
      "        NewOption.value = \"data\";\n",
      "\n",
      "        try\n",
      "        {\n",
      "            view.add(NewOption, null);\n",
      "        }\n",
      "        catch(ex)\n",
      "        {\n",
      "            view.add(NewOption);\n",
      "        }\n",
      "    }\n",
      "  }\n",
      "\n",
      "  function SubmitViewOptionsForm()\n",
      "  {\n",
      "\tvar form = document.forms.ViewOptions;\n",
      "    if(form.form.value == 'html')\n",
      "    {\n",
      "\t\tform.form.setAttribute('disabled','disabled');\n",
      "\t\tif (form.view.value == 'quick') {\n",
      "\t\t\tform.view.setAttribute('disabled','disabled');\n",
      "\t\t}\n",
      "\t\tif (form.targ.value == 'self') {\n",
      "\t\t\tform.targ.setAttribute('disabled','disabled');\n",
      "\t\t}\n",
      "        var token = document.getElementById(\"token_input\");\n",
      "        if (token) {\n",
      "            form.token.value = token.value;\n",
      "        } else {\n",
      "            form.token.setAttribute('disabled','disabled');\n",
      "        }\n",
      "        form.submit();\n",
      "    }\n",
      "    else\n",
      "    {\n",
      "        window.open(\"acc.cgi?acc=\" + form.acc.value + \"&targ=\" + form.targ.value +\n",
      "                  \"&form=\" + form.form.value + \"&view=\" + form.view.value, \"_self\");\n",
      "    }\n",
      "\n",
      "    return false;\n",
      "  }\n",
      "  \n",
      "  function ViewOptionsFormKeyDown(event)\n",
      "  {\n",
      "\tif (event == undefined)\n",
      "\t{    \n",
      "\t\tevent = window.event;\n",
      "\t}\n",
      "\tif (event.keyCode == 13)\n",
      "\t{\n",
      "\t\tSubmitViewOptionsForm();\n",
      "\t\treturn false;\n",
      "\t}\n",
      "  };\n",
      "\n",
      "  function OpenFTP(url)\n",
      "  {\n",
      "    window.open(url.replace('ftp://', 'https://'), '_blank');\n",
      "  }\n",
      "\n",
      "  function OpenLink(url, where)\n",
      "  {\n",
      "    window.open(url, where);\n",
      "  }\n",
      "\n",
      "  utils.addEvent(window, \"load\", OnFormFieldChange)\n",
      "  </script>\n",
      "</link></head>\n",
      "<body background=\"/coreweb/template1/pix/bg_main3.gif\" marginheight=\"20\" topmargin=\"20\">\n",
      "<script src=\"/core/jig/1.15.3/js/jig.min.js\" type=\"text/javascript\"></script>\n",
      "<script src=\"/corehtml/pmc/granthub/v1/granthubsearch.min.js\" type=\"text/javascript\"></script>\n",
      "<script src=\"/geo/js/dd_menu.js\" type=\"text/javascript\"></script>\n",
      "<table align=\"center\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"740\">\n",
      "<tr>\n",
      "<td>\n",
      "<table align=\"center\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"100%\">\n",
      "<tr>\n",
      "<td><a href=\"/\"><img alt=\"NCBI Logo\" border=\"0\" height=\"66\" src=\"/geo/img/ncbi_logo.gif\" width=\"145\"/></a></td>\n",
      "<td align=\"center\" background=\"/coreweb/template1/pix/top_bg_white.gif\" nowrap=\"\" valign=\"middle\" width=\"100%\"><img alt=\"\" border=\"0\" height=\"1\" src=\"/coreweb/template1/pix/pixel.gif\" width=\"550\"/><br/>\n",
      "<a href=\"/geo/\"><img alt=\"GEO Logo\" border=\"0\" src=\"/geo/img/geo_main.gif\"/></a>\n",
      "</td>\n",
      "<td align=\"right\" background=\"/coreweb/template1/pix/top_bg_white.gif\"><img alt=\"\" border=\"0\" height=\"66\" src=\"/coreweb/template1/pix/top_right.gif\" width=\"5\"/></td>\n",
      "</tr>\n",
      "</table>\n",
      "<table align=\"center\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"100%\">\n",
      "<tr>\n",
      "<td><img alt=\"\" border=\"0\" height=\"2\" src=\"/coreweb/template1/pix/top2_left.gif\" width=\"601\"/></td>\n",
      "<td background=\"/coreweb/template1/pix/top2_mid_bg.gif\" width=\"100%\"><img alt=\"\" border=\"0\" height=\"1\" src=\"/coreweb/template1/pix/pixel.gif\" width=\"1\"/></td>\n",
      "<td align=\"right\"><img alt=\"\" border=\"0\" height=\"2\" src=\"/coreweb/template1/pix/top2_right.gif\" width=\"14\"/></td>\n",
      "</tr>\n",
      "</table>\n",
      "<table align=\"center\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\" id=\"galerts_table\" width=\"100%\"></table>\n",
      "<table align=\"center\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"100%\">\n",
      "<tr>\n",
      "<td><img alt=\"\" border=\"0\" height=\"16\" name=\"unl_menu_pix\" src=\"/coreweb/template1/pix/top3_ulm_no_a.gif\" usemap=\"#unlmenu\" width=\"145\"/></td>\n",
      "<td background=\"/coreweb/template1/pix/top3_mainmenu_mid_bg.gif\"><img alt=\"\" border=\"0\" height=\"16\" src=\"/coreweb/template1/pix/top3_mainmenu_left.gif\" width=\"3\"/></td>\n",
      "<td background=\"/coreweb/template1/pix/top3_mainmenu_mid_bg.gif\" nowrap=\"\" valign=\"middle\" width=\"100%\">\n",
      "<!-- GEO Navigation -->\n",
      "<ul id=\"geo_nav_bar\">\n",
      "<li><a href=\"#\">GEO Publications</a>\n",
      "<ul class=\"sublist\">\n",
      "<li><a href=\"/geo/info/GEOHandoutFinal.pdf\">Handout</a></li>\n",
      "<li><a href=\"http://nar.oxfordjournals.org/content/41/D1/D991.full\">NAR 2013 (latest)</a></li>\n",
      "<li><a href=\"http://nar.oupjournals.org/cgi/content/full/30/1/207?ijkey=oxMPOWseARs7o&amp;keytype=ref&amp;siteid=nar\">NAR 2002 (original)</a></li>\n",
      "<li><a href=\"/pmc/3531084,3341798,3013736,2686538,2270403,1669752,1619900,1619899,539976,99122\">All publications</a></li>\n",
      "</ul>\n",
      "</li>\n",
      "<li><a href=\"/geo/info/faq.html\">FAQ</a></li>\n",
      "<li><a href=\"/geo/info/MIAME.html\" title=\"Minimum Information About a Microarray Experiment\">MIAME</a></li>\n",
      "<li><a href=\"mailto:geo@ncbi.nlm.nih.gov\">Email GEO</a></li>\n",
      "</ul>\n",
      "<!-- END GEO Navigation -->\n",
      "</td>\n",
      "<td align=\"right\" background=\"/coreweb/template1/pix/top3_mainmenu_mid_bg.gif\"><img alt=\"\" border=\"0\" height=\"16\" src=\"/coreweb/template1/pix/top3_mainmenu_right.gif\" width=\"5\"/></td>\n",
      "</tr>\n",
      "</table>\n",
      "<table align=\"center\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"100%\">\n",
      "<tr>\n",
      "<td><img alt=\"\" border=\"0\" height=\"4\" src=\"/coreweb/template1/pix/top4_ulm_left.gif\" width=\"145\"/></td>\n",
      "<td background=\"/coreweb/template1/pix/top4_mid_bg.gif\" width=\"100%\"><img alt=\"\" border=\"0\" height=\"1\" src=\"/coreweb/template1/pix/pixel.gif\" width=\"1\"/></td>\n",
      "<td align=\"right\" background=\"/coreweb/template1/pix/top4_mid_bg.gif\"><img alt=\"\" border=\"0\" height=\"4\" src=\"/coreweb/template1/pix/top4_ulm_right.gif\" width=\"5\"/></td>\n",
      "</tr>\n",
      "</table>\n",
      "<table align=\"center\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"100%\">\n",
      "<tr>\n",
      "<td background=\"/coreweb/template1/pix/main_left_bg.gif\" width=\"1\"><img alt=\"\" border=\"0\" height=\"3\" src=\"/coreweb/template1/pix/main_left_bg.gif\" width=\"4\"/></td>\n",
      "<td bgcolor=\"#F0F8FF\" width=\"10000\">\n",
      "<table cellpadding=\"0\" cellspacing=\"0\" width=\"100%\"><tr><td><font class=\"Top_Navigation_text\" color=\"#2F6E87\" face=\"Verdana\" size=\"+1\">   <a href=\"/\">NCBI</a> &gt; <a href=\"/geo\"><font color=\"\">GEO</font></a> &gt; <a href=\"acc.cgi\"><b>Accession Display</b></a><a href=\"javascript:RPopUpWindow_Set(geologinbar_location,260,120,'','','#E1EAE6','','#538AA9','MessageBox2');\" onmouseout=\"RPopUpWindow_Stop()\"><img alt=\"Help\" height=\"11\" src=\"/coreweb/images/long_help4.gif\" style=\"border: none\" width=\"19\"/></a></font></td>\n",
      "<td align=\"right\">Not logged in | <a href=\"/geo/submitter?ix=1H7CUczzDUUcIOTeMDlKyVJrUMYqml2AGubHhlI2u2FVo_4yz9q13Pepanu9VGS333Yi7DZ7rXYcdVdxIxTIuR\">Login</a><a href=\"javascript:RPopUpWindow_Set(geologinbar_login,260,200,'','','#E1EAE6','','#538AA9','MessageBox2');\" onmouseout=\"RPopUpWindow_Stop()\"><img alt=\"Help\" height=\"11\" src=\"/coreweb/images/long_help4.gif\" style=\"border: none\" width=\"19\"/></a></td>\n",
      "</tr></table>\n",
      "</td>\n",
      "<td background=\"/coreweb/template1/pix/main_right_bg.gif\" width=\"1\"><img alt=\"\" border=\"0\" height=\"3\" src=\"/coreweb/template1/pix/main_right_bg.gif\" width=\"4\"/></td>\n",
      "</tr>\n",
      "<tr>\n",
      "<td background=\"/coreweb/template1/pix/main_left_bg.gif\"><img alt=\"\" border=\"0\" height=\"1\" src=\"/coreweb/template1/pix/main_left_bg.gif\" width=\"4\"/></td>\n",
      "<td bgcolor=\"#E0EEEE\" width=\"10000\"><img alt=\"\" border=\"0\" height=\"1\" src=\"/coreweb/template1/pix/pixel.gif\" width=\"1\"/></td>\n",
      "<td align=\"right\" background=\"/coreweb/template1/pix/main_right_bg.gif\"><img alt=\"\" border=\"0\" height=\"1\" src=\"/coreweb/template1/pix/main_right_bg.gif\" width=\"4\"/></td>\n",
      "</tr>\n",
      "<tr>\n",
      "<td background=\"/coreweb/template1/pix/main_left_bg.gif\"><img alt=\"\" border=\"0\" height=\"3\" src=\"/coreweb/template1/pix/main_left_bg.gif\" width=\"4\"/></td>\n",
      "<td bgcolor=\"White\" width=\"100%\">\n",
      "<table align=\"center\" border=\"0\" width=\"98%\">\n",
      "<tr>\n",
      "<td>\n",
      "<table align=\"right\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"100%\">\n",
      "<tr>\n",
      "<td>\n",
      "<script src=\"acc.js\" type=\"text/javascript\"></script>\n",
      "<span id=\"msg_err\" style=\"color:red\"></span>\n",
      "<span id=\"msg_info\" style=\"color:blue\"></span>\n",
      "<table cellpadding=\"0\" cellspacing=\"0\" style=\"border: 1px solid #C0F8FF\"><tr><td><img alt=\" \" height=\"35\" src=\"/coreweb/template1/pix/pixel.gif\" width=\"1\"/></td>\n",
      "<td bgcolor=\"#F0F8FF\" width=\"100%\"><font color=\"#0066CC\" face=\"Arial\" size=\"1\"><div id=\"HelpMessage\" style=\"font: 11px/11px arial, sans-serif\"><strong>GEO help:</strong> Mouse over screen elements for information.</div></font></td>\n",
      "</tr></table>\n",
      "<form action=\"acc.cgi\" enctype=\"application/x-www-form-urlencoded\" id=\"ViewOptions\" method=\"POST\" name=\"ViewOptions\" target=\"_self\"><table border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"100%\"><tr><td></td>\n",
      "<td bgcolor=\"#CCCCCC\" nowrap=\"\" valign=\"middle\" width=\"100%\"><table align=\"left\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\"><tr><td nowrap=\"\"><table border=\"0\" cellpadding=\"0\" cellspacing=\"0\"><tr><td valign=\"middle\"><input id=\"token\" name=\"token\" type=\"hidden\" value=\"\"/><label for=\"scope\">Scope: </label><select id=\"scope\" name=\"targ\" onmouseout=\"onLinkOut('HelpMessage' , geo_empty_help)\" onmouseover=\"onLinkOver('HelpMessage' , geoaxema_scope)\" style=\"font-size: 10px\"><option selected=\"\" value=\"self\">Self</option>\n",
      "<option value=\"gpl\">Platform</option>\n",
      "<option value=\"gsm\">Samples</option>\n",
      "<option value=\"gse\">Series</option>\n",
      "<option value=\"all\">Family</option>\n",
      "</select>\n",
      "  <label for=\"form\">Format: </label><select id=\"form\" name=\"form\" onchange=\"OnFormFieldChange()\" onmouseout=\"onLinkOut('HelpMessage' , geo_empty_help)\" onmouseover=\"onLinkOver('HelpMessage' , geoaxema_format)\" style=\"font-size: 10px\"><option value=\"html\">HTML</option>\n",
      "<option value=\"text\">SOFT</option>\n",
      "<option value=\"xml\">MINiML</option>\n",
      "</select>\n",
      "  <label for=\"view\">Amount: </label><select id=\"view\" name=\"view\" onmouseout=\"onLinkOut('HelpMessage' , geo_empty_help)\" onmouseover=\"onLinkOver('HelpMessage' , geoaxema_amount)\" style=\"font-size: 10px\"><option value=\"brief\">Brief</option>\n",
      "<option selected=\"\" value=\"quick\">Quick</option>\n",
      "</select>\n",
      " <label for=\"geo_acc\">GEO accession: </label><input id=\"geo_acc\" name=\"acc\" onkeydown=\"ViewOptionsFormKeyDown(event)\" onmouseout=\"onLinkOut('HelpMessage' , geo_empty_help)\" onmouseover=\"onLinkOver('HelpMessage' , geoaxema_acc)\" style=\"font-size: 10px\" type=\"text\" value=\"GSM2912912\"/>  </td>\n",
      "<td valign=\"middle\"><img alt=\"Go\" border=\"0\" onclick=\"SubmitViewOptionsForm()\" onmouseout=\"onLinkOut('HelpMessage' , geo_empty_help)\" onmouseover=\"onLinkOver('HelpMessage' , geoaxema_go)\" src=\"/geo/img/buttons/go_button.gif\"/></td>\n",
      "</tr></table></td></tr></table></td>\n",
      "</tr></table></form>\n",
      "<table><tr><td><table cellpadding=\"2\" cellspacing=\"0\" width=\"600\"><tr bgcolor=\"#cccccc\" valign=\"top\"><td colspan=\"2\"><table width=\"600\"><tr><td><strong class=\"acc\" id=\"GSM2912912\"><a href=\"/geo/query/acc.cgi?acc=GSM2912912\" onmouseout=\"onLinkOut('HelpMessage' , geo_empty_help)\" onmouseover=\"onLinkOver('HelpMessage' , geoaxema_recenter)\">Sample GSM2912912</a></strong></td>\n",
      "<td></td>\n",
      "<td align=\"right\" onmouseout=\"onLinkOut('HelpMessage' , geo_empty_help)\" onmouseover=\"onLinkOver('HelpMessage' , geoaxema_gds)\"><a href=\"/gds/?term=GSM2912912[Accession]\">Query DataSets for GSM2912912</a></td>\n",
      "</tr></table></td></tr>\n",
      "<tr valign=\"top\"><td>Status</td>\n",
      "<td>Public on Jan 04, 2021</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Title</td>\n",
      "<td style=\"text-align: justify\">AR14_H3K4me3</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Sample type</td>\n",
      "<td>SRA</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\"> </td>\n",
      "<td></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Source name</td>\n",
      "<td style=\"text-align: justify\">Embryonic stem cells<br/></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Organism</td>\n",
      "<td><a href=\"/Taxonomy/Browser/wwwtax.cgi?mode=Info&amp;id=10090\" onmouseout=\"onLinkOut('HelpMessage' , geo_empty_help)\" onmouseover=\"onLinkOver('HelpMessage' , geoaxema_organismus)\">Mus musculus</a></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Characteristics</td>\n",
      "<td style=\"text-align: justify\">strain: E14<br/>chip antibody: 304M3B-1xHRV3C (lot 10.30.15AG)<br/>amplification cycles: 8<br/>map quality cut-off: 20<br/>mononucleosome cut-off: 200<br/>reference genome: mm10 (GRCm38)<br/></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Treatment protocol</td>\n",
      "<td style=\"text-align: justify\">N/A<br/></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Growth protocol</td>\n",
      "<td style=\"text-align: justify\">Mouse Embryonic Stem Cells (mESC) E14 cell line (129/Ola background) (Hooper et al., 1987) were grown at 37Â°C, 5% CO2, 95% humidity in ES media high glucose DMEM (Invitrogen) supplemented with 15% (v/v) FBS (Gibco), 2mM L-glutamine (Gibco), 1% (v/v) non-essential amino acids (Gibco), 1X Penicillin/Streptomycin (Gibco), 0.1 mM  2-mercaptoethanol (Gibco), 1000U/mL LIF (ESG1107, Millipore), 3 ÂµM CHIR99021 (04-0004, Stemgent), 1 ÂµM PD0325901 (04-0006, Stemgent), sterile filtered 0.1 Âµm membrane, stored in 4Â°C for up to 1 week. E14 cells were cultured on dishes coated with 0.1% gelatin (Sigma) without feeder cells, and passaged daily in 1:3 ratio (media was changed 3 hours prior passage). Cells were harvested at 80-90% confluence, with media changed at least 3 hours prior.<br/></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Extracted molecule</td>\n",
      "<td>genomic DNA</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Extraction protocol</td>\n",
      "<td style=\"text-align: justify\">ICeChIP has been done as previously described (Brand et al., 2008; Grzybowski et al., 2015). Briefly, ~100M plate adherent cells were released using Accutase (Millipore), quenched with complete medium and collected (500rcf, 5 min., 4Â°C). Subsequent steps have been done on ice, cells after each wash were collected by centrifugation (500rcf, 5 min., 4Â°C). Cells were washed twice with 10mL PBS, twice with 5mL of buffer N (15 mM Tris pH 7.5, 15 mM NaCl, 60 mM KCl, 8.5% (w/v) Sucrose, 5 mM MgCl2, 1 mM CaCl2 1 mM DTT, 1x RL Protease Inhibitor Cocktail). Cellâs membrane was lysed by adding 1volume of the 2x Lysis Buffer (Buffer N supplemented with 0.6% NP-40 substitute (Sigma)) to the single cell suspension in 2 PCVs (packed cell volumes) of Buffer N. After 10 minute incubation on ice, nuclei were collected by centrifugation and resuspended in at least 6 PNV (packed nuclei volumes) of buffer N. Subsequently nuclei were layered over 7.5 mL Sucrose Cushion N (15 mM Tris pH 7.5, 15 mM NaCl, 60 mM KCl, 30% (w/v) Sucrose, 5 mM MgCl2, 1 mM CaCl2 1 mM DTT, 1x RL Protease Inhibitor Cocktail, 50 Âµg/mL BSA(NEB)) in a 50 ml centrifuge tube. Nuclei were spun through the sucrose cushion in swinging bucket rotor at 500g for 12 min., 4Â°C. Nuclei were resuspended in 2 PNVs of buffer N, total nuclei acid content was measured spectroscopically at 260nm by Nanodrop (Thermo Scientific) (1A260 = 50 ng/Âµl), prior to measurement, DNA was stripped from chromatin by adding 18ul â 98 ul of 2M NaCl and DNA was fragmented by vortexing and water bath sonication. The amount and quality of nuclei was measured using a hemocytometer. The apparent concentration of chromatin was adjusted to 1ug/ul with Buffer N. The following semi synthetic standards were then spiked in: Symmetrical H3K4me3, H3K9me3, H3K27me3, H3K36me3, H3K79me2, H3K4me3/H3K27me3 cis bivalent, asymmetrical H3K4me3 , H3K4me3/H3K27me3 trans-bivalent. To fragment the DNA, we aliquoted 100 Âµg of chromatin and added 1Worthington unit of Micrococcal nuclease(Worthington) per 4.785Âµg of chromatin (measured at 260nm, 1A260 = 50 ng/Âµl), incubated at 37C for 12 minutes. Digestion was stopped by adding 1/10 volume of 11x MNase stop buffer (110mM EGTA, 110mMEDTA pH 8.0). Nuclei were lysed by slowly adding 5 M NaCl while mixing on a vortex (lowest setting) to the final concentration of 600 mM NaCl. Insoluble material has been spun down at 18000rcf, 1min. 4Â°C.   Hydroxyapatite (HAP) chromatography has been done as described in (Brand et al., 2008). Briefly, 66mg of hydroxyapatite (HAP) resin (Bio-Rad Macro-PrepÂ® Ceramic Hydroxyapatite Type I 20 Î¼m) was rehydrated with 200 Î¼L of HAP buffer 1 (3.42 mM Na2HPO4 and 1.58 mM NaH2PO4 final pH 7.2, 600 mM NaCl, 1 mM EDTA, 200 Î¼M PMSF). Subsequently, 100 Âµg of digested soluble chromatin was added to the rehydrated resin and incubated for 10 minutes at 4C on a rotator. Afterwards, resin slurry was transferred to the centrifugal filter unit (Millipore Ultrafree MCâHV Centrifugal Filter 0.45 Î¼m). Resin was washed 4 times with 200 Î¼l of HAP buffer 1, 4 times with 200 Î¼l of HAP buffer 2 (3.42 mM Na2HPO4 and 1.58 mM NaH2PO4 final pH 7.2, 100 mM NaCl, 1 mM EDTA, 200 Î¼M PMSF), and eluted 3 times with 50 Âµl HAP elution buffer (342 mM Na2HPO4 and 158 mM NaH2PO4 final pH 7.2, 100 mM NaCl, 1 mM EDTA, 200 Î¼M PMSF), each wash/elution step was accompanied by centrifugation step (600rcf, 30s, 4Â°C). Concentration of the chromatin was evaluated spectroscopically by Nanodrop (Thermo Scientific) (1A260 = 50 ng/Âµl). Apparent concentration of the chromatin was adjusted to 20 ng/Âµl with ChIP buffer 1 with BSA (25 mM Tris pH 7.5, 5 mM MgCl2, 100 mM KCl, 10% (v/v) glycerol, 0.1% (v/v) NP-40 substitute, 100Âµg/ml BSA(NEB)).  ICeChIP was performed as previously described (Grzybowski et al., 2015) with following modifications: Î±H3K4me3 ChIP- 5Âµg chromatin, 2Âµg 304M3B-1xHRV3C(Hattori et al., 2016), 40Âµl Streptavidin M-280 Dynabeads (Invitrogen). Î±H3K9me3 ChIP- 3Âµg chromatin, 0.5Âµg 309M3B(Hattori et al., 2016), 10Âµl Streptavidin M-280 Dynabeads (Invitrogen). Î±H3K27me3 ChIP - 0.8Âµg chromatin, 0.6Âµg CST C36B11 lot 8, 5Âµl Protein G Dynabeads (Invitrogen).  Aforementioned volumes of magnetic beads were washed twice with 200Âµl ChIP buffer 1 with BSA. Antibodies were resuspended in 100 Âµl of ChIP buffer 1 with BSA; subsequently, magnetic beads were collected using magnetic rack, supernatant was removed, and magnetic beads were resuspended with the antibody solution and incubated for at least 1hr at 4Â°C on a rotator. Afterwards unbound antibody was washed away with two 200Âµl washes of ChIP buffer 1 with BSA. Streptavidin beads were additionally washed twice with 200Âµl of ChIP buffer 1 with BSA supplemented with 5ÂµM biotin for 10 min at Â°4C on a rotator for each wash, followed by wash with 200Âµl ChIP buffer 1 with BSA. Supernatant was removed on a magnetic rack and specific amount of chromatin, mentioned at the beginning of this chapter, was used to resuspend the magnetic beads. Chromatin was incubated with antibody-beads conjugates for 10-15 minutes at 4Â°C on a rotator.    Subsequently, magnetic beads were washed two times with 200Âµl ChIP buffer 2 (25 mM Tris pH 7.5, 5 mM MgCl2, 300 mM KCl, 10% (v/v) glycerol, 0.1% (v/v) NP-40 substitute, 100 Âµg/ml BSA(NEB)), and one time with 200Âµl ChIP buffer 3(10 mM Tris pH 7.5, 250 mM LiCl, 1mM EDTA, 0.5% Naâ¢Deoxycholate, 0.5%(v/v) NP 40 substitute, 100 Âµg/ml BSA(NEB)), 10 minute, at 4Â°C, on a rotator with tube change after each wash. These washes were followed by quick 200Âµl ChIP buffer 1 (without BSA) wash and 200Âµl TE wash (10mM Tris-HCl pH 8.0, 1mM EDTA). Chromatin was released from the resin using 50Âµl ChIP elution buffer(50 mM Tris pH 7.5, 1 mM EDTA, 1% w/v SDS) at 55Â°C for 5 minutes. Elution was supplemented with 200mM NaCl, 10mM EDTA, 10Âµg of Proteinase K (Roche) and incubated at 55Â°C for 2 hours. DNA was isolated using 3 volumes of Serapure HD (1mg/ml of 1Âµm, hydrophobic, carboxylated, Sera-Mag SpeedBeads (GE 65152105050250), 20% PEG-8000, 2.5 M NaCl, 10 mM Tris pH 7.5, 1 mM EDTA, 0.05% Tween-20, filter sterilized prior to addition of  magnetic beads), incubated for 5 minutes at room temperature, collected with magnetic rack and washed twice with +200Âµl of 75% ethanol, on a magnetic rack without disturbing the magnetic beads. Subsequently, ethanol was carefully removed, and DNA was eluted by resuspending beads in 50Âµl of TE buffer, magnetic beads were collected using magnetic rack and supernatant was moved to a new tube. In order to limit DNA loss, all operations have been performed using 250Âµl siliconized tubes.   Sequential ICeChIP was performed in the same manner as described above with following changes. For the primary IP we have used 5Âµg of HAP purified chromatin, 2Âµg 304M3B-1xHRV3C â HRV3C cleavable, biotinylated, Î±H3K4me3 Fab (Hattori et al., 2016), immobilized on 40Âµl Streptavidin M-280 Dynabeads (Invitrogen). After 10 minutes incubation of chromatin with Fab-resin conjugate, resin was washed three times with 200Âµl ChIP buffer 1 with 100Âµg/ml BSA, each wash consisted of 10 minutes incubation at 4Â°C on a rotator, followed by tube change. Subsequently, resin was briefly washed with 200ul ChIP buffer 1 with 100Âµg/ml BSA, and chromatin was released from the resin with 20Âµl of ChIP buffer 1 supplemented with 100Âµg/ml BSA and 4Âµg HRV3C incubated on ice for 60 minutes, elution step was repeated one more time and both elutions were combined. Primary elution was added to 0.6Âµg CST C36B11, Î±H3K27me3 mAb, immobilized on 5Âµl of Protein G Dynabeads(Invitrogen) for 10 minutes, at 4Â°C, on a rotator. Subsequently, magnetic beads were washed two times with 200Âµl ChIP buffer 2, and one time with 200Âµl ChIP buffer 3, 10 minute, at 4Â°C, on a rotator with tube change after each wash. These washes were followed by quick wash with 200Âµl ChIP buffer 1 (without BSA). Chromatin was released from the resin using 50Âµ<br/></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\"> </td>\n",
      "<td></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Library strategy</td>\n",
      "<td>ChIP-Seq</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Library source</td>\n",
      "<td>genomic</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Library selection</td>\n",
      "<td>ChIP</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Instrument model</td>\n",
      "<td>Illumina HiSeq 4000</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\"> </td>\n",
      "<td></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\">Data processing</td>\n",
      "<td style=\"text-align: justify\">Reference genome was modified to contain sequences of semi-synthetic nucleosome barcodes.<br/>(AR13 only) NEXTflex qRNA-Seq Kit v2 Molecular barcodes were decoded and unique reads were selected using softwere for deconvolution provided by Bioo Scientific. Molecular barcodes were trimmed by removing 12bp from 5' end.<br/>Reads were mapped to GRCm38/mm10+barcodes reference genome, using Bowtie2 (Langmead et al., 2009), end-to-end, sensitive preset.<br/>SAM files were filtered to reject unmapped and unpaired reads, as well as fragments with length &gt; 200bp and Phred quality score &lt; 20.<br/>Paired reads were merged into single interval of the fragment.<br/>BEDTools genomecov(Quinlan and Hall, 2010) waw used to calculate bedgraphs of the genome coverage. BEDTools unionbedg was used to merge IP and input bedgraphs of genome coverage into multiple entry interval file of genome coverage.<br/>Number of DNA fragments coming from semi-synthetic nucleosome standards were counted for each barcode and IP efficiency was calculated for each histone mark.<br/>IP efficiency correction has been applied to calculate HMD as in (Grzybowski et al., 2015).<br/>Genome_build: mm10 (GRCm38)<br/>Supplementary_files_format_and_content: bigwig (Score). Score: HMD - histone modification density; genome_coverage - sequencing depth; transHMD - HMD of transbivalent nucleosomes; HMD_H3K9me3_correction - HMD after correcting for H3K9me3 off-specificity<br/></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\"> </td>\n",
      "<td></td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td>Submission date</td>\n",
      "<td>Jan 04, 2018</td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td>Last update date</td>\n",
      "<td>Jan 05, 2021</td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td>Contact name</td>\n",
      "<td>Alexander Ruthenburg</td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td nowrap=\"\">Organization name</td>\n",
      "<td style=\"text-align: justify\">The University of Chicago<br/></td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td nowrap=\"\">Department</td>\n",
      "<td style=\"text-align: justify\">Molecular Genetics and Cell Biology<br/></td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td nowrap=\"\">Lab</td>\n",
      "<td style=\"text-align: justify\">Ruthenburg Laboratory of Chromatin Biochemistry &amp; Structure<br/></td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td nowrap=\"\">Street address</td>\n",
      "<td style=\"text-align: justify\">920 E 58th St<br/></td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td nowrap=\"\">City</td>\n",
      "<td style=\"text-align: justify\">Chicago</td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td nowrap=\"\">State/province</td>\n",
      "<td style=\"text-align: justify\">Illinois</td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td nowrap=\"\">ZIP/Postal code</td>\n",
      "<td style=\"text-align: justify\">60637</td>\n",
      "</tr>\n",
      "<tr bgcolor=\"#eeeeee\" valign=\"top\"><td nowrap=\"\">Country</td>\n",
      "<td style=\"text-align: justify\">USA</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td nowrap=\"\"> </td>\n",
      "<td></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td>Platform ID</td>\n",
      "<td><a href=\"/geo/query/acc.cgi?acc=GPL21103\">GPL21103</a></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td>Series (1)</td>\n",
      "<td onmouseout=\"onLinkOut('HelpMessage' , geo_empty_help)\" onmouseover=\"onLinkOver('HelpMessage' , geoaxema_recenter)\"><table cellpadding=\"3\" style=\"position:relative;top:-5px;left:-5px\"><tr><td valign=\"top\"><a href=\"/geo/query/acc.cgi?acc=GSE108747\" onmouseout=\"onLinkOut('HelpMessage' , geo_empty_help)\" onmouseover=\"onLinkOver('HelpMessage' , geoaxema_recenter)\">GSE108747</a></td>\n",
      "<td valign=\"top\">H3K4me3/H3K27me3 bivalent chromatin function to fine-tune gene expression level</td>\n",
      "</tr></table></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td colspan=\"2\"><strong>Relations</strong></td></tr>\n",
      "<tr valign=\"top\"><td>BioSample</td>\n",
      "<td><a href=\"https://www.ncbi.nlm.nih.gov/biosample/SAMN08291596\">SAMN08291596</a></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td>SRA</td>\n",
      "<td><a href=\"https://www.ncbi.nlm.nih.gov/sra?term=SRX3531930\">SRX3531930</a></td>\n",
      "</tr>\n",
      "</table>\n",
      "<br/><table cellpadding=\"2\" cellspacing=\"2\" width=\"600\"><tr bgcolor=\"#eeeeee\" valign=\"top\"><td align=\"middle\" bgcolor=\"#CCCCCC\"><strong>Supplementary file</strong></td>\n",
      "<td align=\"middle\" bgcolor=\"#CCCCCC\"><strong>Size</strong></td>\n",
      "<td align=\"middle\" bgcolor=\"#CCCCCC\"><strong>Download</strong></td>\n",
      "<td align=\"middle\" bgcolor=\"#CCCCCC\"><strong>File type/resource</strong></td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td bgcolor=\"#DEEBDC\">GSM2912912_AR14_H3K4me3_mQ20_L200_HMD.bigwig</td>\n",
      "<td bgcolor=\"#DEEBDC\" title=\"3988436124\">3.7 Gb</td>\n",
      "<td bgcolor=\"#DEEBDC\"><a href=\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2912nnn/GSM2912912/suppl/GSM2912912%5FAR14%5FH3K4me3%5FmQ20%5FL200%5FHMD%2Ebigwig\">(ftp)</a><a href=\"/geo/download/?acc=GSM2912912&amp;format=file&amp;file=GSM2912912%5FAR14%5FH3K4me3%5FmQ20%5FL200%5FHMD%2Ebigwig\">(http)</a></td>\n",
      "<td bgcolor=\"#DEEBDC\">BIGWIG</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td bgcolor=\"#EEEEEE\">GSM2912912_AR14_H3K4me3_mQ20_L200_HMD_H3K9me3_correction.bigwig</td>\n",
      "<td bgcolor=\"#EEEEEE\" title=\"4212724466\">3.9 Gb</td>\n",
      "<td bgcolor=\"#EEEEEE\"><a href=\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2912nnn/GSM2912912/suppl/GSM2912912%5FAR14%5FH3K4me3%5FmQ20%5FL200%5FHMD%5FH3K9me3%5Fcorrection%2Ebigwig\">(ftp)</a><a href=\"/geo/download/?acc=GSM2912912&amp;format=file&amp;file=GSM2912912%5FAR14%5FH3K4me3%5FmQ20%5FL200%5FHMD%5FH3K9me3%5Fcorrection%2Ebigwig\">(http)</a></td>\n",
      "<td bgcolor=\"#EEEEEE\">BIGWIG</td>\n",
      "</tr>\n",
      "<tr valign=\"top\"><td bgcolor=\"#DEEBDC\">GSM2912912_AR14_H3K4me3_mQ20_L200_genome_coverage.bigwig</td>\n",
      "<td bgcolor=\"#DEEBDC\" title=\"442849279\">422.3 Mb</td>\n",
      "<td bgcolor=\"#DEEBDC\"><a href=\"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2912nnn/GSM2912912/suppl/GSM2912912%5FAR14%5FH3K4me3%5FmQ20%5FL200%5Fgenome%5Fcoverage%2Ebigwig\">(ftp)</a><a href=\"/geo/download/?acc=GSM2912912&amp;format=file&amp;file=GSM2912912%5FAR14%5FH3K4me3%5FmQ20%5FL200%5Fgenome%5Fcoverage%2Ebigwig\">(http)</a></td>\n",
      "<td bgcolor=\"#DEEBDC\">BIGWIG</td>\n",
      "</tr>\n",
      "<tr><td><a href=\"/Traces/study/?acc=SRX3531930\">SRA Run Selector</a><a href=\"javascript:RPopUpWindow_Set(geoaxema_srarun,260,120,'','','#E1EAE6','','#538AA9','MessageBox2');\" onmouseout=\"RPopUpWindow_Stop()\"><img alt=\"Help\" height=\"11\" src=\"/coreweb/images/long_help4.gif\" style=\"border: none\" width=\"19\"/></a></td></tr>\n",
      "<tr><td class=\"message\">Raw data are available in SRA</td></tr>\n",
      "<tr><td class=\"message\">Processed data provided as supplementary file</td></tr>\n",
      "</table>\n",
      "<span id=\"customDlArea\"></span><br/></td></tr></table>\n",
      "</td>\n",
      "</tr>\n",
      "</table>\n",
      "</td>\n",
      "</tr>\n",
      "</table>\n",
      "</td>\n",
      "<td background=\"/coreweb/template1/pix/main_right_bg.gif\"><img alt=\"\" border=\"0\" height=\"3\" src=\"/coreweb/template1/pix/main_right_bg.gif\" width=\"4\"/></td>\n",
      "</tr>\n",
      "<tr>\n",
      "<td background=\"/coreweb/template1/pix/but_left.gif\"><img alt=\"\" border=\"0\" height=\"4\" src=\"/coreweb/template1/pix/but_left.gif\" width=\"4\"/></td>\n",
      "<td background=\"/coreweb/template1/pix/but_mid_bg.gif\" bgcolor=\"#FFFFFF\" width=\"10000\"><img alt=\"\" border=\"0\" height=\"1\" src=\"/coreweb/template1/pix/pixel.gif\" width=\"1\"/></td>\n",
      "<td align=\"right\" background=\"/coreweb/template1/pix/but_right.gif\"><img alt=\"\" border=\"0\" height=\"4\" src=\"/coreweb/template1/pix/but_right.gif\" width=\"4\"/></td>\n",
      "</tr>\n",
      "</table>\n",
      "<table align=\"center\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"100%\">\n",
      "<tr>\n",
      "<td width=\"99%\"><img alt=\"\" border=\"0\" height=\"1\" src=\"/coreweb/template1/pix/pixel.gif\" width=\"1\"/></td><td align=\"right\" nowrap=\"\" valign=\"top\">\n",
      "<span class=\"HELPBAR\">|<a href=\"https://www.nlm.nih.gov\"> NLM </a>|<a class=\"HELPBAR\" href=\"https://www.nih.gov\"> NIH </a>|<a class=\"HELPBAR\" href=\"mailto:geo@ncbi.nlm.nih.gov\"> GEO Help </a>|<a class=\"HELPBAR\" href=\"/geo/info/disclaimer.html\"> Disclaimer </a>|<a class=\"HELPBAR\" href=\"https://www.nlm.nih.gov/accessibility.html\"> Accessibility </a>|</span><br/>\n",
      "</td>\n",
      "</tr>\n",
      "</table>\n",
      "<map name=\"unlmenu\">\n",
      "<area alt=\"NCBI Home\" coords=\"2,0,39,15\" href=\"/\" onmouseout=\"changpics(unl_menu_pix, unl_menu_noa)\" onmouseover=\"changpics(unl_menu_pix, unl_menu_home_a)\"/>\n",
      "<area alt=\"NCBI Search\" coords=\"40,0,91,15\" href=\"/ncbisearch/\" onmouseout=\"changpics(unl_menu_pix, unl_menu_noa)\" onmouseover=\"changpics(unl_menu_pix, unl_menu_search_a)\"/>\n",
      "<area alt=\"NCBI SiteMap\" coords=\"92,0,143,15\" href=\"/Sitemap/\" onmouseout=\"changpics(unl_menu_pix, unl_menu_noa)\" onmouseover=\"changpics(unl_menu_pix, unl_menu_sitemap_a)\"/>\n",
      "</map>\n",
      "<script src=\"/portal/portal3rc.fcgi/rlib/js/InstrumentNCBIBaseJS/InstrumentPageStarterJS.js\" type=\"text/javascript\"> </script>\n",
      "</td></tr></table></body>\n",
      "</html>\n",
      "\n",
      "[]\n",
      "[]\n"
     ]
    }
   ],
   "source": [
    "# Место для решения\n",
    "\n",
    "import os\n",
    "path_to_code = os.getcwd()\n",
    "path_to_inp_file = os.path.join(path_to_code, os.pardir, 'lections', 'files_and_images', 'gsm_list.txt')\n",
    "\n",
    "with open(path_to_inp_file, 'r') as inp_file:\n",
    "    list_of_id = []\n",
    "    for line in inp_file:\n",
    "        line = line.rstrip()\n",
    "        list_of_id.append(line)\n",
    "list_of_id\n",
    "\n",
    "import json\n",
    "import requests\n",
    "\n",
    "#response = requests.get('https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2912912')\n",
    "response = requests.get(f'https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc={list_of_id[0]}')\n",
    "#print(response.text)\n",
    "\n",
    "import bs4\n",
    "#list_of_html = []\n",
    "response_to_bs = bs4.BeautifulSoup(response.text, \"html.parser\")\n",
    "print(response_to_bs)\n",
    "list_of_p = response_to_bs('p')\n",
    "print(list_of_p)\n",
    "list_of_series = response_to_bs('Series')\n",
    "print(list_of_series)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Задача 2\n",
    "В файле encode_id_list.txt содержатся идентификаторы базы данных ENCODE (идентификаторы с префиксом ENCSR). Вам необходимо подгрузить json с информацией по каждому эксперименту. Например, https://www.encodeproject.org/experiments/ENCSR747VUU/?format=json. Обратите внимание на то, как устроена URL-ссылка. Ознакомтесь с JSON'ом и имеющейся в нём информацией. Напишите программу, которая собирает информацию об типе эксперимента (assay_title), таргете (target -> label) и дате сабмита данных (date_submitted) для каждого идентификатора. Результат представьте в виде таблицы, состоящей из 4 колонок."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Место для решения"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
